### PR TITLE
Blob cache API in Platform

### PR DESCRIPTION
**Pull request by pixelflinger**
This is compatible with EGL_ANDROID_blob_cache.

### CODE REVIEW

File: filament/backend/include/backend/Platform.h [hunk 1]
```diff
@@ -79,6 +82,81 @@ class UTILS_PUBLIC Platform {
      * thread, or if the platform does not need to perform any special processing.
      */
     virtual bool pumpEvents() noexcept;
+
+    /**
+     * InsertBlobFunc is an Invocable to an application-provided function that a
+     * backcend implementation may use to insert a key/value pair into the
```
Comment (bejado) [inline filament/backend/include/backend/Platform.h:88]:
```suggestion
     * backend implementation may use to insert a key/value pair into the
```

**Review by bejado — APPROVED** [review]

File: filament/backend/include/backend/Platform.h [hunk 2]
```diff
@@ -79,6 +82,81 @@ class UTILS_PUBLIC Platform {
...
+     * recently set value for a given key.
+     *
+     * @param key           pointer to the beginning of the key data that is to be inserted
+     * @param keySize       specifies the size in byte of the data pointed to by <key>
+     * @param value         pointer to the beginning of the value data that is to be inserted
+     * @param valueSize     specifies the size in byte of the data pointed to by <value>
+     */
+    void insertBlob(const void* key, size_t keySize, const void* value, size_t valueSize);
+
+     /**
```
Comment (poweifeng) [inline filament/backend/include/backend/Platform.h:138]: nit: comment misaligned by one space?

**Review by poweifeng — APPROVED** [review]

File: filament/include/filament/Material.h [hunk 3]
```diff
@@ -150,6 +152,27 @@ class UTILS_PUBLIC Material : public FilamentAPI {
...
+     * Asynchronously ensures that a subset of this Material's variants are compiled. After issuing
+     * several Material::compile() calls in a row, it is recommended to call Engine::flush()
+     * such that the backend can start the compilation work as soon as possible.
+     * The provided callback is guaranteed to be called on the main thread after all unfiltered
+     * variants of the material are compiled. This can take hundreds of milliseconds.
+     *
+     * If all the material's variants are already compiled, the callback will be scheduled as
+     * soon as possible, but this might take a few dozen millisecond, corresponding to how
+     * many previous frames are enqueued in the backend. This also varies by backend. Therefore,
+     * it is commanded to only call this method once per material shortly after creation.
```
Comment (poweifeng) [inline filament/include/filament/Material.h:165]: recommended

**Review by poweifeng — APPROVED** [review]

File: filament/include/filament/Material.h (hunk 3 above)
Comment (poweifeng) [inline filament/include/filament/Material.h:165]:
```suggestion
     * it is recommended to only call this method once per material shortly after creation.
```

**Review by poweifeng — APPROVED** [review]

**Pull request merged**